android/src/main/java/io/invertase/firebase/links/RNFirebaseLinks.java

```diff
@@ -85,7 +108,7 @@ public void onActivityResult(Activity activity, int requestCode, int resultCode,
 
   @Override
   public void onNewIntent(Intent intent) {
-    // TODO: Do I need to re-register the links handler for each new intent?
+    handleLink();
```

Comment:
Shouldn't we use the new `intent` data?

Reply:
You're right. The strange thing is this is somehow works. I'll take a look again later. You can also try to check it with dynamic_link branch..

Reply:
It was fixed. BTW, usually it won't even look at the intent data - and will take the link from the capture data in the server. see: https://firebase.google.com/docs/reference/android/com/google/firebase/dynamiclinks/FirebaseDynamicLinks.html#getDynamicLink(android.content.Intent)